controllers/admin/AdminStatusesController.php

```diff
@@ -622,7 +622,10 @@ public function postProcess()
             if (!$order_state->isRemovable()) {
                 $this->errors[] = $this->trans('For security reasons, you cannot delete default order statuses.', [], 'Admin.Shopparameters.Notification');
             } else {
-                return parent::postProcess();
+                $order_state->deleted = 1;
```

Comment:
Yes, I thought it would add overall complexity to have sometimes a soft delete and sometimes a hard delete, do you think it's worth it?

Reply:
It's not mandatory for sure, but once we validate and release this behaviour we won't be able to change it back without a BC Maybe this should be discussed with @PrestaShop/prestashop-product-team

Reply:
i see 2 potential problems/risks : 
- can it cause a performance problem if there are thousands / hundreds of thousands order statuses in the database with soft delete status ?
- what happens if i try to create an order status that i had decided to delete in the past ? Will i get an error or is it possible to "undelete" an order that was soft deleted ?

Reply:
1- I think there can only be drawbacks in BO as in any other parts the soft deleted states are ignored
2- if you deleted one and recreate it this will be a new record

Reply:
So if we always soft delete them regardless of their use they will keep piling up But then maybe this should be check directly in `OrderState::delete` method, so the behaviour will remain the same no matter where it's called (a module, the legacy controller, the migrated page)

Reply:
@jolelievre @colinegin 

order states are so rarely deleted that it shouldn't be an issue here, if we have `softDelete` method available from 1.7.7 we can - as a developers - think about proper indexes if we need more performance :) but it's not something what should be done in core IMO

Reply:
thank you for your input @kpodemski ! you're right, and it has been deleted that it will be soft delete only.